flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1203,12 +1203,17 @@ void cancelOrFailAndCancelInvokableInternal(ExecutionState targetState, Throwabl
                 return;
             }
 
-            if (current == ExecutionState.DEPLOYING || current == ExecutionState.CREATED) {
+            if (current == ExecutionState.CREATED) {
                 if (transitionState(current, targetState, cause)) {
                     // if we manage this state transition, then the invokable gets never called
                     // we need not call cancel on it
                     return;
                 }
+            } else if (current == ExecutionState.DEPLOYING) {
+                if (transitionState(current, targetState, cause)) {
```

Comment:
It's better to add some comments:

Tasks may hang on the Invokable constructor or static code. We need add a watchdog to ensure the task does not remain hanging.

Reply:
Thx, updated